source/serialize.hpp

```diff
@@ -37,22 +38,22 @@ concept is_metric_like = requires(T t) {
 };
 
 template<is_metric_like T>
-constexpr auto serialize(const T& metric, const Tags& global_tags) -> std::string
+BARK_CONSTEXPR auto serialize(const T& metric, const Tags& global_tags) -> std::string
 {
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wfloat-equal"
     auto out = fmt::memory_buffer {};
     // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
     if constexpr (has_sample_rate<T>) {
-        out.reserve(metric.metric.size() + 1 +                               // metric_name:
-                    23 +                                                     // value|<indicator>
-                    (metric.sample_rate != 1.0 ? 5 : 0) +                    // |@<sample_rate>
-                    metric.tags.str().size() + global_tags.str().size() + 2  // |#<tags>
+        out.try_reserve(metric.metric.size() + 1 +                               // metric_name:
```

Comment:
`reserve` invokes `try_reserve` as the only thing.

`reserve` is not constexpr.
`try_reserve` is constexpr...